pkg/storage/storagewrappers/combinedtuplereader_test.go

```diff
@@ -0,0 +1,831 @@
+			gotArr := make([]*openfgav1.Tuple, 0)
+			for {
+				if tuple, err := got.Next(tt.args.ctx); err == nil {
+					gotArr = append(gotArr, tuple)
+				} else {
+					break
+				}
+			}
+			if !reflect.DeepEqual(gotArr, tt.want) {
+				t.Errorf("ReadStartingWithUser() got = %v, want %v", gotArr, tt.want)
+			}
+		})
+	}
+}
+
+func Test_combinedTupleReader_ReadUserTuple(t *testing.T) {
+	mockCtl, mockRelationshipTupleReader := makeMocks(t)
+	defer mockCtl.Finish()
+
+	type fields struct {
+		RelationshipTupleReader storage.RelationshipTupleReader
+		contextualTuples        []*openfgav1.TupleKey
+	}
+	type args struct {
+		ctx     context.Context
+		store   string
+		tk      *openfgav1.TupleKey
+		options storage.ReadUserTupleOptions
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		setups  func()
+		want    *openfgav1.Tuple
+		wantErr bool
+	}{
+		{
+			name: "Test combinedTupleReader ReadUserTuple OK - contextual tuple found",
+			fields: fields{
+				RelationshipTupleReader: mockRelationshipTupleReader,
+				contextualTuples: []*openfgav1.TupleKey{
+					testTuples["group:1#member@user:11"].GetKey(),
+					testTuples["group:1#member@user:12"].GetKey(),
+				},
+			},
+			setups: func() {
+				mockRelationshipTupleReader.
+					EXPECT().
+					ReadUserTuple(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
+					Times(0)
+			},
+			args: args{
+				ctx:   context.Background(),
+				store: "1",
+				tk: &openfgav1.TupleKey{
+					User:     "user:11",
+					Relation: "member",
+					Object:   "group:1",
+				},
+				options: storage.ReadUserTupleOptions{},
+			},
+			want:    testTuples["group:1#member@user:11"],
+			wantErr: false,
+		},
+		{
+			name: "Test combinedTupleReader ReadUserTuple OK - contextual tuple not found",
+			fields: fields{
+				RelationshipTupleReader: mockRelationshipTupleReader,
+				contextualTuples: []*openfgav1.TupleKey{
+					testTuples["group:1#member@user:11"].GetKey(),
+					testTuples["group:1#member@user:12"].GetKey(),
+				},
+			},
+			setups: func() {
+				mockRelationshipTupleReader.
+					EXPECT().
+					ReadUserTuple(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
+					Return(testTuples["group:1#member@user:13"], nil)
+			},
+			args: args{
+				ctx:   context.Background(),
+				store: "1",
+				tk: &openfgav1.TupleKey{
+					User:     "user:13",
+					Relation: "member",
+					Object:   "group:1",
+				},
+				options: storage.ReadUserTupleOptions{},
+			},
+			want:    testTuples["group:1#member@user:13"],
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.setups != nil {
+				tt.setups()
+			}
+			c := NewCombinedTupleReader(tt.fields.RelationshipTupleReader, tt.fields.contextualTuples)
+
+			got, err := c.ReadUserTuple(tt.args.ctx, tt.args.store, tt.args.tk, tt.args.options)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ReadUserTuple() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
```

Comment:
```suggestion
			if tt.wantErr {
				require.Error(t, err)
				return
			}
```

Reply:
Could you simplify this/make it more readable?

`if (err != nil) != tt.wantErr {`